pam_mysql.c

```diff
@@ -3023,6 +3122,30 @@ static pam_mysql_err_t pam_mysql_check_passwd(pam_mysql_ctx_t *ctx,
 #endif
             } break;
 
+          case 7:
+            {
+              /* Salted SHA */
+#ifdef HAVE_PAM_MYSQL_SHA1_DATA
+              unsigned char* hash;
+              size_t sha1_size;
+              Base64Decode(row[0], &hash, &sha1_size);
+              size_t salt_length = sha1_size - 20;
+              unsigned char salt[salt_length];
+              memcpy(salt, &(hash[20]), salt_length);
+
+              char buf[41];
+              pam_mysql_ssha_data((unsigned char*)passwd, strlen(passwd), salt, salt_length,
+                  buf);
+              vresult = strcmp(row[0], buf);
+              {
+                char *p = buf - 1;
+                while (*(++p)) *p = '\0';
+              }
+#else
+              syslog(LOG_AUTHPRIV | LOG_ERR, PAM_MYSQL_LOG_PREFIX "non-crypt()ish SSHA hash is not supported in this build.");
+#endif
+            } break;
+
```

Comment:
This is analogous to SHA1. Please provide an example of the test scenario for SHA1, so I can customize it for SSHA.